test/api/test_tools.py

```diff
@@ -1867,6 +1867,23 @@ def test_map_over_collection_structured_like(self):
             assert second_collection_level['collection_type'] == 'list'
             assert second_collection_level['elements'][0]['element_type'] == 'hda'
 
+    @skip_without_tool("__SPLIT_COLLECTION_BY_NUMBER__")
+    def test_split_collection_by_number(self):
+        with self.dataset_populator.test_history() as history_id:
+            hdca_id = self.dataset_collection_populator.create_list_in_history(history_id, contents=[("A", "A"), ("B", "B")]).json()['id']
+            self.dataset_populator.wait_for_history(history_id, assert_ok=True)
+            inputs = {
+                "input": {'values': [dict(src="hdca", id=hdca_id)]},
+                "how|how_split": "number_of_collections",
+                "how|num_collecton": 2
+            }
+            generated_collections = self._run("__FILTER_FROM_FILE__", history_id, inputs, assert_ok=True)['generated_collections']
+            output_collection_1, output_collection_2 = generated_collections
+            self.dataset_populator.wait_for_history(history_id, assert_ok=True)
```

Comment:
This wait doesn't do anything, that's implied by `assert_ok=True`
```suggestion
```

Reply:
I think its something to do with the test framework as all I have is this:
```
def test_split_collection_by_number(self):
        with self.dataset_populator.test_history() as history_id:
            hdca_id = self.dataset_collection_populator.create_list_in_history(history_id, contents=["A", "A", "B", "B"]).json()['id']
            self.dataset_populator.wait_for_history(history_id, assert_ok=True)
            inputs = {
                "input": {'values': [dict(src="hdca", id=hdca_id)]},
                "how|how_split": "number_of_collections",
                "how|num_collecton": {'values': 2}
            }
            generated_collections = self._run("__SPLIT_COLLECTION_BY_NUMBER__", history_id, inputs, assert_ok=True)['output_collections']
```
And the error is coming from _patch_library_inputs as:
```
File "/home/CSCScience.ca/dhole/repos/galaxy/lib/galaxy/webapps/galaxy/api/tools.py", line 576, in _patch_library_inputs
    for index, value in enumerate(v['values']):
TypeError: 'int' object is not iterable
```
The value of 2 is getting input correctly, it just won't let it go through it seems. Making it a list doesn't work in any capacity either from all the ways I've tried it.

Thanks for all the suggestions though, I'll keep playing around with it.

Reply:
I can also do a PR against your branch, the tests are passing for me with the suggested changes.

Reply:
Have a look at https://github.com/phac-nml/galaxy/pull/4 which fixes the API test

Reply:
I've realized that I've had num_collecton instead of num_collection this whole time. 

Thanks for the help at it